src/lib/snapper/snapshot_dbus.rb

```diff
@@ -0,0 +1,214 @@
+require 'yast'
+require 'dbus'
+
+module Yast2
+  module Snapper
+    class SnapshotDBus
+      
+      include Yast::Logger
+
+      TYPE_INT_TO_SYMBOL = {
+        0 => :SINGLE,
+        1 => :PRE,
+        2 => :POST
+      }
+      def initialize(context)
+        @context = context
+      end
+
+      class << self
+        
+        def all(configs)
+          configs = list_configs if !configs
+          snapshots = [configs].flatten.each_with_object([]) do |config, a|
```

Comment:
why add configs to array and then flatten it? it smell to me like handling single config and not array and in such case, there is better methods like `Array`